compiler/src/dotty/tools/dotc/sbt/ExtractDependencies.scala

```diff
@@ -165,6 +165,30 @@ private class ExtractDependenciesCollector(rec: DependencyRecorder) extends tpd.
         rec.addClassDependency(parent.tpe.classSymbol, depContext)
     }
 
+  // Only reference DependencyByMacroExpansion if it an be found on the classpath,
+  // as it was added later to the zinc.apiinfo DependencyContext enum
+  // e.g. pre 1.10.x sbt would throw java.lang.NoSuchFieldError errors here
+  lazy val allowsDependencyByMacroExpansion =
+    classOf[DependencyContext].getFields().exists(_.getName() == "DependencyByMacroExpansion")
+
+  private def addMacroDependency(sym: Symbol)(using Context): Unit =
+    if (allowsDependencyByMacroExpansion) {
+      if (!ignoreDependency(sym)) {
+        if (!sym.is(Package)) {
+          val enclOrModuleClass = if (sym.is(ModuleVal)) sym.moduleClass else sym.enclosingClass
+          assert(enclOrModuleClass.isClass, s"$enclOrModuleClass, $sym")
+
+          rec.addClassDependency(enclOrModuleClass, DependencyByMacroExpansion)
+        }
+      }
+    }
+
+  private def addMacroDependency(trees: List[Tree])(using Context): Unit =
+    val traverser = new TypeDependencyTraverser {
+      def addDependency(symbol: Symbol) = addMacroDependency(symbol)
+    }
+    trees.foreach(tree => traverser.traverse(tree.tpe))
```

Comment:
Looks like this whole function should be elided when the macro dependency tracking is unavailable